CI build macos-latest with libsodium compilation failed

# Issue description
  CXX      unittests/unittest_curve_encoding-unittest_curve_encoding.o
In file included from ../../unittests/unittest_curve_encoding.cpp:10:
../../src/curve_mechanism_base.hpp:9:10: fatal error: 'sodium.h' file not found
  CXXLD    unittests/unittest_ypipe
#include "sodium.h"
         ^~~~~~~~~~
  CXXLD    unittests/unittest_ip_resolver
1 error generated.
make[3]: *** [unittests/unittest_curve_encoding-unittest_curve_encoding.o] Error 1
make[3]: *** Waiting for unfinished jobs....
  CXXLD    unittests/unittest_udp_address
ld: warning: -bind_at_load is deprecated on macOS
ld: warning: search path '/Users/runner/work/libzmq/libzmq/libzmq/tmp/lib' not found
ld: warning: -bind_at_load is deprecated on macOS
ld: warning: search path '/Users/runner/work/libzmq/libzmq/libzmq/tmp/lib' not found
ld: warning: -bind_at_load is deprecated on macOS
ld: warning: search path '/Users/runner/work/libzmq/libzmq/libzmq/tmp/lib' not found
make[2]: *** [check-am] Error 2
make[1]: *** [check-recursive] Error 1
make: *** [distcheck] Error 1

More details: https://github.com/zeromq/libzmq/actions/runs/10122411078/job/27994528728

--- a/Makefile.am
+++ b/Makefile.am
@@ -1418,6 +1418,11 @@ unittests_unittest_curve_encoding_LDADD =  \
         $(top_builddir)/src/.libs/libzmq.a \
         ${src_libzmq_la_LIBADD} \
         $(CODE_COVERAGE_LDFLAGS)
+
+if USE_LIBSODIUM
+unittests_unittest_curve_encoding_CPPFLAGS += ${sodium_CFLAGS}
+unittests_unittest_curve_encoding_LDADD += ${sodium_LIBS}
+endif
 endif
 
 check_PROGRAMS = ${test_apps}

